semcore/d3-chart/src/RadialTree.tsx

```diff
@@ -0,0 +1,768 @@
+import React from 'react';
+import { Component, sstyled, CProps, ReturnEl } from '@semcore/core';
+import createElement from './createElement';
+import uniqueIDEnhancement from '@semcore/utils/lib/uniqueID';
+import { transition } from 'd3-transition';
+import style from './style/radial-tree.shadow.css';
+import { shade } from '@semcore/utils/lib/color';
+
+const memoize = <T extends (...args: (string | number)[]) => any>(func: T): T => {
+  const results: { [cacheKey: string]: ReturnType<T> } = {};
+  return ((...args) => {
+    const argsKey = args.join(',');
+    if (!results[argsKey]) {
+      results[argsKey] = func(...args);
+    }
+    return results[argsKey];
+  }) as T;
+};
+
+const measureText = memoize((text: string, fontSize: number) => {
+  const span = document.createElement('span');
+  span.style.fontSize = `${fontSize}px`;
+  for (const line of text.split('\n')) {
+    span.append(document.createTextNode(line));
+    span.append(document.createElement('br'));
+  }
+  span.style.display = 'inline-block';
+  document.body.append(span);
+  const textWidth = span.offsetWidth;
+  span.remove();
+  return textWidth;
+});
+
+const baseAngle = -Math.PI / 2; // The top vertical line
+
+type RadianData = {
+  /**
+   * Text label on the end of radian.
+   */
+  label: string;
+  /**
+   * Radian key. Radian, which `key` is equal to `activeKey` is displayed as active.
+   */
+  key: string;
```

Comment:
key prop use React, maybe this key will be mislead developers

Reply:
`id`, `name`? If you can suggest some better one, it's ok for me to change prop name but for now `key` seems to be the most suitable for me